autorest/azure/environments.go

```diff
@@ -133,12 +140,37 @@ var (
 	}
 )
 
-// EnvironmentFromName returns an Environment based on the common name specified
+// EnvironmentFromName returns an Environment based on the common name specified.
 func EnvironmentFromName(name string) (Environment, error) {
+	// IMPORTANT
+	// As per @radhikagupta5:
+	// This is technical debt, fundamentally here because Kubernetes is not currently accepting
+	// contributions to the providers. Once that is an option, the provider should be updated to
+	// directly call `EnvironmentFromFile`. Until then, we rely on dispatching Azure Stack environment creation
+	// from this method based on the name that is provided to us.
+	if strings.EqualFold(name, "AZURESTACKCLOUD") {
+		return EnvironmentFromFile(os.Getenv(EnvironmentFilepathName))
+	}
+
 	name = strings.ToUpper(name)
 	env, ok := environments[name]
 	if !ok {
 		return env, fmt.Errorf("autorest/azure: There is no cloud environment matching the name %q", name)
 	}
+
 	return env, nil
 }
+
+// EnvironmentFromFile loads an Environment from a configuration file available on disk.
+// This function is particularly useful in the Hybrid Cloud model, where one must define their own
+// endpoints.
+func EnvironmentFromFile(location string) (unmarshaled Environment, err error) {
```

Comment:
Would it make sense to add an example in here on the file format,
azurestackcloud.json:
{
   "name": "AzureStackCloud",
    "managementPortalURL": "",
    "publishSettingsURL": "",
    "serviceManagementEndpoint": "https://management.azurestackci13.onmicrosoft.com/e55c980f-9a38-48ab-a059-87027f8134b1",
    "resourceManagerEndpoint": "https://management.redmond.ext-u15e0303.masd.stbtest.microsoft.com",
    "activeDirectoryEndpoint": "https://login.windows.net/",
    "galleryEndpoint": "https://portal.redmond.ext-u15e0303.masd.stbtest.microsoft.com:30015/",
    "keyVaultEndpoint": "https://vault.redmond.ext-u15e0303.masd.stbtest.microsoft.com",
    "graphEndpoint": "https://graph.windows.net/",
    "storageEndpointSuffix": "redmond.ext-u15e0303.masd.stbtest.microsoft.com",
    "sQLDatabaseDNSSuffix": "",
    "trafficManagerDNSSuffix": "",
    "keyVaultDNSSuffix": "vault.redmond.ext-u15e0303.masd.stbtest.microsoft.com",
    "serviceBusEndpointSuffix": "",
    "serviceManagementVMDNSSuffix": "cloudapp.net",
    "resourceManagerVMDNSSuffix": "cloudapp.azure.com",
    "containerRegistryDNSSuffix": "azurecr.io"
 }

Reply:
Or in a separate README kind of file, not sure.